src/test/run-make/issue-11908/Makefile

```diff
@@ -13,11 +13,11 @@ all:
 	mv $(call DYLIB,foo) $(TMPDIR)/other
 	$(RUSTC) foo.rs --crate-type=dylib
 	$(RUSTC) bar.rs -L $(TMPDIR)/other 2>&1 | \
-		grep "multiple dylib candidates"
+		grep "assertion failed" && exit 1 || exit 0
```

Comment:
Ok to take care of those you can just compile all dylibs here with `-C prefer-dynamic`